internal/command/deploy/strategy_bluegreen.go

```diff
@@ -734,7 +736,7 @@ func getZombies(ids map[string]bool) (map[string]bool, error) {
 
 	sort.Ints(numbers)
 
-	delete(ids, fmt.Sprint(numbers[0]))
+	delete(ids, fmt.Sprint(numbers[len(numbers)-1]))
```

Comment:
Previously, it would only retain the oldest deployment. However, there's two cases where this is a problem:

1. The deployment finished except for a partial blue destroy step. For example, we could end up with only one machine in the oldest deployment.

2. The machine list contains a machine without the `fly_bluegreen_deployment_tag` so the code defaults it to -1 and it's the only machine that will be retained.